GTFO-API/API/LocalizationAPI.cs

```
@@ -0,0 +1,722 @@
+    }
+
+    internal static void OnGameDataInitialized()
+    {
+        s_GameDataInitialized = true;
+        foreach (string entry in s_EntriesToGenerateTextDBs)
+        {
+            if (!s_Entries.TryGetValue(entry, out LocalizationEntry? localizationEntry))
+            {
+                continue;
+            }
+            localizationEntry.GenerateTextDataBlock(entry, force: true);
+        }
+    }
+
+    internal static void OnGameAssetsLoaded()
+    {
+        // current language should be available now.
+        Status.Ready = true;
+    }
+
+    internal static void LanguageChanged()
+    {
+        OnLanguageChange?.Invoke();
+    }
+
+    /// <summary>
+    /// An event called when the game's current language is changed.
+    /// </summary>
+    public static event Action? OnLanguageChange;
+
+    /// <summary>
+    /// Gets the current language GTFO is using.
+    /// </summary>
+    public static Language CurrentLanguage
+        => Text.TextLocalizationService.CurrentLanguage;
+
+    /// <summary>
+    /// Get a localization string using the specified key for the current language,
+    /// and formats it with the specified <paramref name="args"/>, or if not found
+    /// returns <paramref name="key"/>.
+    /// </summary>
+    /// <param name="key">
+    /// The case-sensitive localization entry key.
+    /// </param>
+    /// <param name="args">
+    /// The arguments to format the string with.
+    /// </param>
+    /// <returns>
+    /// The found localization value for the current language formatted with
+    /// <paramref name="args"/>, or <paramref name="key"/>.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="key"/> is <see langword="null"/>.
+    /// </exception>
+    public static string FormatString(string key, params object?[] args)
+    {
+        ValidateLocalizationKey(key);
+
+        if (!TryGetString(key, out string? value))
+        {
+            return key;
```

I think it's better throw exception here instead of just returning key value